intellij/src/saros/intellij/eventhandler/filesystem/LocalFilesystemModificationHandler.java

```diff
@@ -640,6 +650,31 @@ private void generateFolderMoveActivity(
     }
   }
 
+  /**
+   * Returns whether the given virtual file is a content root for the module it is associated with.
+   *
+   * @param virtualFile the virtual file to check
+   * @return <code>true</code> when the virtual file is one of the content roots of the module it is
+   *     associated with, <code>false</code> if it represents a file, does not match one of the
+   *     content roots or is not associated with a module
+   */
+  private boolean isContentRootDirectory(@NotNull VirtualFile virtualFile) {
+    if (!virtualFile.isDirectory()) {
+      return false;
+    }
+
+    Module module =
+        Filesystem.runReadAction(() -> ModuleUtil.findModuleForFile(virtualFile, project));
```

Comment:
Nit: Eclipse creates a warning: "The static method should be accessed directly" and would like you to use `ModuleUtilCore.findModuleForFile(virtualFile, project));`

Reply:
I am not sure whether the read access is strictly necessary here, but I just wanted to use it to make sure. The call `ModuleUtilCore.findModuleForFile(virtualFile, project))` internally just calls `ProjectFileIndex.SERVICE.getInstance(project).getModuleForFile(file)` which, according to its javadoc, should always be called from a read action. The `ModuleUtilCore` however does not honor this restriction.